core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -137,7 +138,13 @@ private[spark] class EventLoggingListener(
     // scalastyle:on println
     if (flushLogger) {
       writer.foreach(_.flush())
-      hadoopDataStream.foreach(_.hflush())
+      hadoopDataStream.foreach(ds => {
```

Comment:
hsync() is even stronger than hflush(), since under the cover both methods use the same flushOrSync(), but hsync performs an additional tasks like flushing OS buffers (fsync).